src/pages/Main.tsx

```diff
@@ -0,0 +1,38 @@
+import { Component, type ReactNode } from 'react';
+import Search from '../components/Search';
+import ErrorBoundary from '../components/ErrorBoundary';
+import CardList from '../components/CardList';
+import { SEARCH_KEY } from '../utils/constants';
+
+interface State {
+  searchText: string;
+}
+
+class Main extends Component<Record<string, never>, State> {
+  constructor(props: Record<string, never>) {
+    super(props);
+    this.state = {
+      searchText: localStorage.getItem(SEARCH_KEY) || '',
```

Comment:
```suggestion
      searchText: localStorage.getItem(SEARCH_KEY) ?? '',
```
for fallback with null or undefine better "??" - it makes the code more expressive